lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1004,105 +1011,103 @@ export class UserAgentApplication {
   acquireTokenSilent(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
     return new Promise<string>((resolve, reject) => {
       // Validate scopes
-      // TODO: Replace with new validation pattern
-      const isValidScope = this.validateInputScope(scopes);
-      if (isValidScope && !Utils.isEmpty(isValidScope)) {
+      try {
+        scopes = this.validateAndFilterInputScope(scopes);
+      } catch (err) {
+        if (err instanceof ClientConfigurationError) {
+          // These will return errors instead of strings in the future
+          // Should we throw instead of reject here?
+          reject(err.message);
+        } else {
+          reject(ErrorCodes.inputScopesError + Constants.resourceDelimeter + ErrorDescription.inputScopesError);
+        }
+      }
+
+      const scope = scopes.join(" ").toLowerCase();
+      const userObject = user ? user : this.getUser();
+      const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+      //if user is not currently logged in and no login_hint/sid is passed as an extraQueryParamater
+      if (!userObject && Utils.checkSSO(extraQueryParameters) && Utils.isEmpty(adalIdToken) ) {
+        this._logger.info("User login is required");
         // TODO: Reject with custom error here
-        // TODO: Is this always accessToken?
-        reject(ErrorCodes.inputScopesError + "|" + isValidScope);
+        reject(ErrorCodes.userLoginError + Constants.resourceDelimeter + ErrorDescription.userLoginError);
```

Comment:
same comment here about interp / factor common strings you are using

Reply:
Refactoring, please check new commit